core/controllers/profile.py

```diff
@@ -243,6 +245,12 @@ def get(self):
         """Handles GET requests."""
         return_url = str(self.request.get('return_url', self.request.uri))
 
+        # Check if the return url is the creator dashboard. If it is, we should
+        # set the default dashboard of the user as the creator dashboard.
+        if feconf.CREATOR_DASHBOARD_URL in return_url:
```

Comment:
I think you should do this in a later stage, only after the user has actually signed up. In general, avoid state-changing operations in GET requests.